src/cljs/org/broadinstitute/firecloud_ui/main.cljs

```diff
@@ -333,31 +361,14 @@
                                (swap! state assoc :config-status :error)))}))
    :authenticate-user
    (fn [{:keys [this state]}]
-     (let [hash (nav/get-hash-value)
-           at-index (utils/str-index-of hash "?access_token=")
-           [_ token] (when-not (neg? at-index) (clojure.string/split (subs hash at-index) #"="))
-           attempt-auth (fn [token on-done]
-                          ;; Use basic ajax call here to bypass default ajax-orch failure behavior.
-                          (utils/ajax {:url (str (config/api-url-root) "/me")
-                                       :headers {"Authorization" (str "Bearer " token)}
-                                       :on-done on-done}))]
-       (if-not (neg? at-index)
-         (do
-           ;; New access token. Attempt to authenticate with it.
-           (.replace (.-location js/window) (str "#" (subs hash 0 at-index)))
-           (attempt-auth token (fn [{:keys [success? status-code]}]
-                                 (cond
-                                   (= 401 status-code)
-                                   (swap! state assoc :user-status :auth-failure)
-                                   (= 403 status-code)
-                                   (do (utils/set-access-token-cookie token)
-                                       (swap! state assoc :user-status :not-activated))
-                                   ;; 404 just means not registered
-                                   (or success? (= status-code 404))
-                                   (do (utils/set-access-token-cookie token)
-                                       (react/call :handle-successful-auth this token))
-                                   :else
-                                   (swap! state assoc :user-status :error)))))
+     (if-let [parent (.-opener js/window)]
+       ;; This window was used for login. Send the token to the parent. No need to render the UI.
+       (let [parsed (parse-hash-into-map)]
+         (.postMessage parent (clj->js parsed) (.. js/window -location -origin))
```

Comment:
can you use window.opener instead of postMessage, eliminating the use of postMessage entirely? Since window.opener has inherent same-origin policy enforced by the JavaScript VM, it's the stronger security option.